lib/active_record/connection_adapters/cockroachdb_adapter.rb

```diff
@@ -255,6 +235,21 @@ def self.database_exists?(config)
         false
       end
 
+      def initialize(...)
+        super
+
+      # This rescue flow appears in new_client, but it is needed here as well
+      # since Cockroach will sometimes not raise until a query is made.
+      rescue ActiveRecord::StatementInvalid => error
+        no_db_err_check1 = @connection_parameters && @connection_parameters[:dbname] && error.cause.message.include?(@connection_parameters[:dbname])
+        no_db_err_check2 = @connection_parameters && @connection_parameters[:dbname] && error.cause.message.include?("pg_type")
+        if no_db_err_check1 || no_db_err_check2
+          raise ActiveRecord::NoDatabaseError
+        else
+          raise ActiveRecord::ConnectionNotEstablished, error.message
+        end
+      end
```

Comment:
This is also outdated compared to the rails db, and not tested in our database, and I don't know if _since Cockroach will sometimes not raise until a query is made._ is still valid and what if refers to. Do you have an idea of how to reproduce this behaviour @rafiss ?

Reply:
I am not sure, but a guess I have is that it refers to the fact that CRDB allows you to connect even if the connection string refers to a database name that does not exist. But when AR tries using that database name to do something, CRDB will return an error.